docs/proposals-accepted/202209-receive-tenant-external-labels.md

```diff
@@ -0,0 +1,52 @@
+---
+type: proposal
+title: Allow specifying tenant-specific external labels in Receivers
+status: accepted
+owner: haanhvu
+menu: proposals-accepted
+---
+
+## 1 Related links/tickets
+
+* https://github.com/thanos-io/thanos/issues/5434
+
+## 2 What problem(s) are we trying to solve?
+
+We want to add arbitrary external labels to each tenant in Thanos Receiver.
+
+## 3 How is it going to be helpful?
+
+This new functionality is helpful in many use cases. Two typical ones are:
+* Compact tenant blocks based on tenants’ external labels
+* Query data from tenants which share the same external labels
```

Comment:
Does the existing `tenant_id` work in this case? I understand the second query usecase as users may want to have some cross-tenants aggregation queries or query tenants with some other labels like `region`.

For the first usecase, when do we want to compact based on other external labels? I think we should always compact using `tenant_id`, right?

Reply:
If you do some query based on some other external label, and you know that you'll do that query a lot, would compact based on that external label make your query faster? That's my understanding.

Reply:
Compact blocks means merging blocks together to a single block. Each tenant's blocks should only merge with their own blocks. I don't think it is correct to merge blocks across tenants.
I think what you want is to have some kind of secondary index to group some tenants blocks together? That's just logical grouping, right? Not compaction.

Reply:
The querier can filter out stores which do not match certain external labels, so maybe we need to rephrase this to say:
* Reduce fanout when querying data from tenants which share certain external labels